src/services/dynamo-database-service.ts

```diff
@@ -39,62 +39,23 @@ export class DynamoDatabaseService {
    *
    * @param userId - the userId that comes from the request
    */
-  public async retrieveRecordsByUserId(userId: string) {
-    logger.debug(`${LOGS_PREFIX_SENSITIVE_INFO} Attempting request to dynamo db, with ID : ${userId}`);
-    const parameters: QueryCommandInput = {
-      TableName: this.tableName,
-      KeyConditionExpression: '#pk = :id_value',
-      ExpressionAttributeNames: { '#pk': 'pk' },
-      ExpressionAttributeValues: { ':id_value': { S: userId } },
-      ProjectionExpression: 'blocked, suspended, resetPassword, reproveIdentity, sentAt, appliedAt, isAccountDeleted',
-    };
-
+  public async getAccountStateInformation(userId: string) {
+    const parameters = this.getInputParameterForDatabaseQuery(userId);
+    parameters.ProjectionExpression =
+      'blocked, suspended, resetPassword, reproveIdentity, sentAt, appliedAt, isAccountDeleted';
     const response: QueryCommandOutput = await this.dynamoClient.send(new QueryCommand(parameters));
-    if (!response.Items) {
-      const errorMessage = 'DynamoDB may have failed to query, returned a null response.';
-      logger.error(errorMessage);
-      logAndPublishMetric(MetricNames.DB_QUERY_ERROR_NO_RESPONSE);
-      throw new Error(errorMessage);
-    }
-
-    if (response.Items.length > 1) {
-      const errorMessage = 'DynamoDB returned more than one element.';
-      logger.error(errorMessage);
-      logAndPublishMetric(MetricNames.DB_QUERY_ERROR_TOO_MANY_ITEMS);
-      throw new TooManyRecordsError(errorMessage);
-    }
-    return response.Items[0] ? (unmarshall(response.Items[0]) as DynamoDBStateResult) : undefined;
+    return this.validateQueryResponse(response) as DynamoDBStateResult;
```

Comment:
super minor point, this whole method should return DynamoDBStateResult | undefined. 
So I would change this to:
`const response = this.validateQueryResponse(response)
return response? response as DynamoDBStateResult :  undefined`
So that the compiler knows that this method can return undefined